Makefile

```diff
@@ -28,10 +28,14 @@ BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 # You can use it as an arg. (E.g make bundle-build BUNDLE_IMG=<some-registry>/<project-name-bundle>:<tag>)
 BUNDLE_IMG ?= controller-bundle:$(VERSION)
 
+# Set the shell used to bash for better error handling.
+SHELL = /bin/bash
+.SHELLFLAGS = -e -o pipefail -c
+
 # Image URL to use all building/pushing image targets
 IMG ?= controller:latest
-# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+# Produce CRDs that work with Kubernetes 1.16+ and support defaulting, version conversion, and field pruning
```

Comment:
Are we planning to require 1.16 at a minimum?

Reply:
I would say "yes" because of the following reasons:

- the current version is 1.20 and providers like EKS and the like only maintain support for 3-4 older versions
- 1.16 is stable and has been out for quite a while so programming to this version seems like a reasonable middle ground as opposed to supporting something like 1.12
- 1.16 gives use really good support for CRD versioning.

Reply:
I suspect this will cause issues for a number of customers. I'm aware of some that have only just updated to 1.15.

Reply:
Platform has used new features to push clients to upgrade Kubernetes (for their benefit) in the past; it often works. I'm going to program with the expectation that we're running on 1.16+ and we'll figure out some graceful degradation path if and when.